packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -110,15 +125,45 @@ class GameWidget<T extends Game> extends StatefulWidget {
     this.focusNode,
     this.autofocus = true,
     MouseCursor? mouseCursor,
-  }) : super(key: key) {
+  })  : create = null,
+        super(key: key) {
     if (mouseCursor != null) {
-      game.mouseCursor = mouseCursor;
+      game!.mouseCursor = mouseCursor;
     }
     if (initialActiveOverlays != null) {
-      game.overlays.addAll(initialActiveOverlays);
+      game!.overlays.addAll(initialActiveOverlays);
     }
   }
 
+  /// Renders a [game] in a flutter widget tree.
+  ///
+  /// Unlike the default constructor, this creates a [GameWidget] that controls
+  /// the creation of the game instance, removing the necessity of rendering
+  /// this widget inside a [StatefulWidget].
+  ///
+  /// Ex:
+  /// ```
+  /// ...
+  /// Widget build(BuildContext  context) {
+  ///   return GameWidget.controlled(
+  ///     create: MyGameClass.new,
+  ///   )
+  /// }
+  /// ...
+  /// ```
```

Comment:
It's still a bit unclear about the lifetime of the instance created. For example, what happens to the instance when I replace `GameWidget(game: game)` with `GameWidget(create: () => MyGame())`? Or when `GameWidget(create: () => MyGame1())` is replaced with `GameWidget(create: () => MyGame2())`?

Reply:
Gonna put that in the docs, but basically, to recreate the game instance, one has to use keys to reset the element (consequently the state).

This property defines a routine method that is not executed on every build but is declarable inside a build method.
A more orthodox (but yet more complex) API for this practice is to use delegates. It is standard in the framework, ex: CustomPaint, CustomSingleChildLayout, SliverGrid. All have delegates which are classes that define routine methods that are not executed in every build. They also usually have a filter method that defines if the main methods should be called in a given build phase. For example, `CustomPainter` has a `shouldRepaint` method.

 I know that the current proposal is not the most orthodox way of doing things but I agree with Erick's approach since it makes the API way more succinct and easy to understand. A simple doc explanation will do.